process/coordinator/process.go

```diff
@@ -518,8 +518,10 @@ func (tc *transactionCoordinator) CreateMbsAndProcessCrossShardTransactionsDstMe
 		return miniBlocks, nrTxAdded, false, nil
 	}
 
-	crossMiniBlockHashes := hdr.GetMiniBlockHeadersWithDst(tc.shardCoordinator.SelfId())
-	for key, senderShardId := range crossMiniBlockHashes {
+	shouldSkipShard := make(map[uint32]bool)
```

Comment:
this function needs a unit test to see how the new code works.

Reply:
It is tested through all the relayed integration tests which now pass on each run :)

Reply:
Will add a unit test anyway